rustls/src/lib.rs

```diff
@@ -409,6 +414,7 @@ pub use crate::tls12::Tls12CipherSuite;
 pub use crate::tls13::Tls13CipherSuite;
 pub use crate::verify::DigitallySignedStruct;
 pub use crate::versions::{SupportedProtocolVersion, ALL_VERSIONS, DEFAULT_VERSIONS};
+#[cfg(feature = "webpki")]
```

Comment:
It doesn't seem like `RootCertStore` and `TrustAnchorWithDn` depend on webpki?

Reply:
`TrustAnchorWithDn` could live outside of `crate::webpki` now, if it would be useful independent of webpki.

`RootCertStore::add` and `RootCertStore::add_parsable_certificates` both depend on `webpki::extract_trust_anchor`, and without those functions it seems like `RootCertStore` is a more directly written `Vec<TrustAnchorWithDn>`?

Reply:
Yup, fair point about the dependency on `webpki::extract_trust_anchor`! I guess one more general question I have is, why are you making webpki optional in the first place? It doesn't seem necessary for making *ring* optional...

Reply:
Mmm, it is in fact completely unnecessary now webpki no longer implies a fundamental dependency on *ring*. Going to drop this commit entirely.